hash/hash.go

```diff
@@ -0,0 +1,101 @@
+// Package hash provides utility functions to process complex data types, like
+// data streams, files, or sequences of structs of different types.
+package hash
+
+import (
+	"bytes"
+	"errors"
+	h "hash"
+	"io"
+	"os"
+
+	"encoding"
+	"reflect"
+)
+
+var defaultChunkSize = 1024
```

Comment:
IIRC I wrote that thing, so I take most of the blame for this:
- why `Stream` and `File` with the same function-signature?
- why chunking on your own in `StreamChunk` if you can have `io.Copy`?

I liked those functions back then, but more I know about hashing, the less I like those, because it hides a lot of complexity that is source of problems if you want to do the hashing in a cross-language way...

I would keep:
`Stream` that uses `io.Copy` to make a hash of a stream
`File` as convenience function that opens a file and calls `Stream` on it
`Args` (can we find a better name?), even though I'm not that convinced anymore...